openml/runs/functions.py

```diff
@@ -270,8 +270,7 @@ def run_flow_on_task(  # noqa: C901, PLR0912, PLR0915, PLR0913
             raise PyOpenMLError(
                 "Flow does not exist on the server, but 'flow.flow_id' is not None."
             )
-
-        if upload_flow and flow_id is None:
+        if upload_flow and flow_id is False:
```

Comment:
`flow_exists` does not return `None` but `False` when a flow does not exist.
I find `flow_id is False` easier to parse than `not flow_id` since `flow_id` can be both integer or boolean. While both are correct, the check is clearer to me this way.

Reply:
This bug would fail a unit test only the first time around, unit tests would later upload the flow anyway which means the second time this path is no longer taken and tests can pass. Because there are other tests which update server state that need to be ran a few times, this error was never caught as it was just in the initial sea of reds after a server reset.